Modules/ZDC/src/ZDCRawDataCheck.cxx

```diff
@@ -64,20 +88,128 @@ std::string ZDCRawDataCheck::getAcceptedType() { return "TH1"; }
+      msg->Draw();
     } else if (checkResult == Quality::Medium) {
-      ILOG(Info, Support) << "Quality::medium, setting to orange" << ENDM;
+      std::string errorSt = "Warning baseline value in the channels: " + mStringW + "is not correct. Send mail to the expert";
+      TLatex* msg = new TLatex(mPosMsgX, mPosMsgY, errorSt.c_str());
+      msg->SetNDC();
+      msg->SetTextSize(16);
+      msg->SetTextFont(43);
+      msg->SetTextColor(kOrange);
+      h->GetListOfFunctions()->Add(msg);
       h->SetFillColor(kOrange);
+      msg->Draw();
     }
     h->SetLineColor(kBlack);
   }
 }
 
+void ZDCRawDataCheck::init()
+{
+  mVectParam.clear();
+  setChName("ZNAC");
+  setChName("ZNA1");
+  setChName("ZNA2");
+  setChName("ZNAS");
+  setChName("ZNA3");
+  setChName("ZNA4");
+  setChName("ZNCC");
+  setChName("ZNC1");
+  setChName("ZNC2");
+  setChName("ZNCS");
+  setChName("ZNC3");
+  setChName("ZNC4");
+  setChName("ZPAC");
+  setChName("ZEM1");
+  setChName("ZPA1");
+  setChName("ZPA2");
+  setChName("ZPAS");
+  setChName("ZPA3");
+  setChName("ZPA4");
+  setChName("ZPCC");
+  setChName("ZEM2");
+  setChName("ZPC3");
+  setChName("ZPC4");
+  setChName("ZPCS");
+  setChName("ZPC1");
+  setChName("ZPC2");
+  for (int i = 0; i < (int)mVectParam.size(); i++) {
+    setChCheck(i);
+  }
+  if (const auto param = mCustomParameters.find("POS_MSG_X"); param != mCustomParameters.end()) {
+    mPosMsgX = atof(param->second.c_str());
+  }
+  if (const auto param = mCustomParameters.find("POS_MSG_Y"); param != mCustomParameters.end()) {
+    mPosMsgY = atof(param->second.c_str());
+  }
+}
+
+void ZDCRawDataCheck::setChName(std::string channel)
+{
+  sCheck chCheck;
+  chCheck.ch = channel;
+  mVectParam.push_back(chCheck);
+}
+
+void ZDCRawDataCheck::setChCheck(int index)
+{
+  std::vector<std::string> tokenString;
+  if (const auto param = mCustomParameters.find(mVectParam.at(index).ch); param != mCustomParameters.end()) {
+    tokenString = tokenLine(param->second, ";");
+    mVectParam.at(index).minW = atof(tokenString.at(0).c_str()) - atof(tokenString.at(1).c_str());
+    mVectParam.at(index).maxW = atof(tokenString.at(0).c_str()) + atof(tokenString.at(1).c_str());
+    mVectParam.at(index).minE = atof(tokenString.at(0).c_str()) - atof(tokenString.at(2).c_str());
+    mVectParam.at(index).maxE = atof(tokenString.at(0).c_str()) + atof(tokenString.at(2).c_str());
+  }
+}
+
+std::vector<std::string> ZDCRawDataCheck::tokenLine(std::string Line, std::string Delimiter)
+{
+  std::string token;
+  size_t pos = 0;
+  int i = 0;
+  std::vector<std::string> stringToken;
+  while ((pos = Line.find(Delimiter)) != std::string::npos) {
+    token = Line.substr(i, pos);
+    stringToken.push_back(token);
+    Line.erase(0, pos + Delimiter.length());
+  }
+  stringToken.push_back(Line);
+  return stringToken;
+}
+
+void ZDCRawDataCheck::dumpVecParam(int numBinHisto, int num_ch)
+{
+  std::ofstream dumpFile;
+  dumpFile.open("dumpStructuresRawCheck.txt");
```

Comment:
I see this is commented out, but please do not assume in general that you will be able to open a file, could you add a check before you write to it?